File: src/programmers/p1/P1_12940.java
package programmers.p1;

import java.util.Arrays;

public class P1_12940 {
    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution(6, 8)));
    }

    public static int[] solution(int n, int m) {
        return new int[]{gcd(n, m), lcm(n, m)};
    }

    public static int lcm(int n, int m) {
        int min = Math.min(n, m);
        int max = Math.max(n, m);
        int lcm = 1;
        for (int i = 2; i <= min; ) {
            if (min % i == 0 && max % i == 0) {
                min /= i;
                max /= i;
                lcm *= i;
            } else
                i++;
        }
        lcm = lcm * min * max;
        return lcm;
    }

    public static int gcd(int a, int b) {
        if (b == 0)
            return a;
        else
            return gcd(b, a % b);
    }
}

File: src/programmers/p1/P1_12940.kt
package programmers


fun main() {
    println(solution(6, 8).contentToString())
    println(solution(3, 12).contentToString())
    println(solution(2, 5).contentToString())
}

private fun solution(n: Int, m: Int): IntArray = intArrayOf(gcd(n, m), lcm(n, m))

private fun lcm(a: Int, b: Int): Int = a * b / gcd(a, b)

private fun gcd(a: Int, b: Int): Int = if (b == 0) a else gcd(b, a % b)
